### PR TITLE
chore: merge release/1.0 into develop

### PR DESCRIPTION
**Pull request by jonathan-teh**

### CODE REVIEW

**Comment by jonathan-teh** [comment]
bors try

**Review by tiagolobocastro — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,14 +12,14 @@ repos:
     -   id: rust-style
         name: Rust style
         description: Run cargo fmt on files included in the commit. rustfmt should be installed before-hand.
-        entry: cargo +nightly-2021-11-30 fmt --all -- --check
+        entry: cargo fmt --all -- --check
```
Comment (akhilerm) [inline .pre-commit-config.yaml:15]: This change was introduced in the [PR](https://github.com/openebs/mayastor/pull/1052) to make use of `spdk-rs` crate. Wont switching it back cause issues in those set of commits in develop?

File: chart/templates/csi-daemonset.yaml [hunk 2]
```diff
@@ -43,7 +43,9 @@ spec:
             fieldRef:
               fieldPath: status.podIP
         - name: RUST_BACKTRACE
-          value: "1"
+          value: "1"{{ if .Values.moac }}
+        - name: MOAC
+          value: "true"{{ end }}
```
Comment (akhilerm) [inline chart/templates/csi-daemonset.yaml:48]:
```suggestion
          value: "1"
{{ if .Values.moac }}
        - name: MOAC
          value: "true"
{{ end }}
```

`if` in helm can be written in the following format to make it more readable.

File: deploy/csi-daemonset.yaml [hunk 3]
```diff
@@ -30,7 +30,7 @@ spec:
       # the same.
       containers:
       - name: mayastor-csi
-        image: mayadata/mayastor-csi:develop
+        image: mayadata/mayastor:develop
```
Comment (akhilerm) [inline deploy/csi-daemonset.yaml:33]: Do we run mayastor-csi also from the mayastor-binary?

**Review by akhilerm — COMMENTED** [review]
A few questions on the charts and usage of rust nightly

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jonathan-teh) [reply .pre-commit-config.yaml:15]: No, this was previously reverted in cfe04ce1, it mainly affects those (like me) who don't use the `rustup` toolchain. The `+xyz` bit actually applies to `rustup` and so doesn't work if you don't have it installed. Without that, it simply runs the current default toolchain.

File: chart/templates/csi-daemonset.yaml (hunk 2 above)
Comment (jonathan-teh) [reply chart/templates/csi-daemonset.yaml:48]: This is a merge request, I only went as far as making it build and pass pre-commit and CI tests. Feel free to raise another PR to fix this.

File: deploy/csi-daemonset.yaml (hunk 3 above)
Comment (jonathan-teh) [reply deploy/csi-daemonset.yaml:33]: It would appear so. This and the previous comment refer to code originally part of #1045.

**Review by mtzaurus — APPROVED** [review]

**Comment by jonathan-teh** [comment]
bors merge

**Pull request merged**